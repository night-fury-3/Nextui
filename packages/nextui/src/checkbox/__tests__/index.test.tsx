import React from 'react';
import { mount, ReactWrapper, render } from 'enzyme';
import Checkbox from '../index';

const getCheckboxElement = (wrapper: ReactWrapper) => {
  return wrapper.find('[type="checkbox"]');
};

const expectCheckboxIsChecked = (wrapper: ReactWrapper, value: boolean) => {
  expect(getCheckboxElement(wrapper).props().checked).toBe(value);
};

const expectCheckboxToHaveARIAChecked = (
  wrapper: ReactWrapper,
  value: boolean
) => {
  expect(getCheckboxElement(wrapper).props()['aria-checked']).toBe(value);
};

const expectCheckboxToHaveARIADisabled = (
  wrapper: ReactWrapper,
  value: boolean
) => {
  expect(getCheckboxElement(wrapper).props()['aria-disabled']).toBe(value);
};

describe('Checkbox', () => {
  it('should render correctly', () => {
    const wrapper = mount(<Checkbox checked={true}>Buenos Aires</Checkbox>);
    expect(() => wrapper.unmount()).not.toThrow();
    const rendered = render(<Checkbox>Buenos Aires</Checkbox>);
    expect(rendered.html()).toMatchSnapshot();
  });

  it('should work correctly with different sizes', () => {
    const wrapper = mount(
      <div>
        <Checkbox size="xs">mini</Checkbox>
        <Checkbox size="sm">small</Checkbox>
        <Checkbox size="md">medium</Checkbox>
        <Checkbox size="lg">large</Checkbox>
        <Checkbox size="xl">xlarge</Checkbox>
      </div>
    );
    expect(wrapper.html()).toMatchSnapshot();
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it('should work correctly with initial value', () => {
    let wrapper = mount(<Checkbox checked={true}>Buenos Aires</Checkbox>);
    expectCheckboxIsChecked(wrapper, true);

    wrapper = mount(<Checkbox checked={false}>Buenos Aires</Checkbox>);
    expectCheckboxIsChecked(wrapper, false);

    wrapper = mount(<Checkbox initialChecked>Buenos Aires</Checkbox>);
    expectCheckboxIsChecked(wrapper, true);

    wrapper = mount(<Checkbox initialChecked={false}>Buenos Aires</Checkbox>);
    expectCheckboxIsChecked(wrapper, false);

    expect(() => wrapper.unmount()).not.toThrow();
  });

  it('should change value after click', () => {
    const Wrapper = () => {
      const [state, setState] = React.useState<string>('state1');

      return (
        <Checkbox initialChecked onChange={() => setState('state2')}>
          {state}
        </Checkbox>
      );
    };
    const wrapper = mount(<Wrapper />);
    const input = getCheckboxElement(wrapper);
    input.simulate('change');
    expect(wrapper.find('.text').text()).toContain('state2');
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it('should ignore events when disabled', () => {
    const Wrapper = () => {
      const [state, setState] = React.useState<string>('state1');

      return (
        <Checkbox disabled onChange={() => setState('state2')}>
          {state}
        </Checkbox>
      );
    };
    const wrapper = mount(<Wrapper />);
    const input = getCheckboxElement(wrapper);
    input.simulate('change');
    expect(wrapper.find('.text').text()).not.toContain('state2');
    expectCheckboxIsChecked(wrapper, false);
    expectCheckboxToHaveARIAChecked(wrapper, false);
    expectCheckboxToHaveARIADisabled(wrapper, true);
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it('should work correctly with indeterminate value', () => {
    let wrapper = mount(<Checkbox indeterminate>Buenos Aires</Checkbox>);
    let icon = wrapper.find('i').getDOMNode();
    expect((icon as HTMLInputElement).className).toContain('indeterminate');

    wrapper = mount(<Checkbox indeterminate={false}>Buenos Aires</Checkbox>);
    icon = wrapper.find('i').getDOMNode();
    expect((icon as HTMLInputElement).className).not.toContain('indeterminate');

    expect(() => wrapper.unmount()).not.toThrow();
  });

  it('should work correctly with line-through value', () => {
    let wrapper = mount(<Checkbox line>Buenos Aires</Checkbox>);
    let icon = wrapper.find('.text').getDOMNode();
    expect((icon as HTMLInputElement).className).toContain('line-through');

    wrapper = mount(<Checkbox line={false}>Buenos Aires</Checkbox>);
    icon = wrapper.find('.text').getDOMNode();
    expect((icon as HTMLInputElement).className).not.toContain('line-through');

    expect(() => wrapper.unmount()).not.toThrow();
  });
});
