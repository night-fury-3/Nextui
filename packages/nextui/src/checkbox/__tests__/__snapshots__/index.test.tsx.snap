// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render correctly 1`] = `"<div class=\\"nextui-c-hGXOEk nextui-c-huiNHE nextui-c-hGXOEk-jewhUT-color-default nextui-checkbox-container\\" tabindex=\\"0\\"><input type=\\"checkbox\\" class=\\"nextui-c-VVylU nextui-checkbox-input\\" tabindex=\\"-1\\" data-state=\\"uncheked\\" aria-checked=\\"false\\" aria-disabled=\\"false\\"><div class=\\"nextui-c-fwBnkL nextui-checkbox-mask\\"><i class=\\"nextui-c-iQLINm nextui-c-iQLINm-hwMtuN-indeterminate-false nextui-c-iQLINm-gXZkKf-size-md nextui-icon-check\\"><div class=\\"nextui-c-edsSUO nextui-icon-check-line1\\"></div><div class=\\"nextui-c-khOxv nextui-icon-check-line2\\"></div></i></div></div><span class=\\"nextui-c-HDlzY nextui-c-HDlzY-cyFBRX-color-default nextui-checkbox-text\\">Buenos Aires</span>"`;

exports[`Checkbox should work correctly with different sizes 1`] = `"<div><label class=\\"nextui-c-gLvohg nextui-c-gLvohg-eWknYe-size-xs nextui-checkbox-label\\"><div class=\\"nextui-c-hGXOEk nextui-c-huiNHE nextui-c-hGXOEk-jewhUT-color-default nextui-checkbox-container\\" tabindex=\\"0\\"><input type=\\"checkbox\\" class=\\"nextui-c-VVylU nextui-checkbox-input\\" tabindex=\\"-1\\" data-state=\\"uncheked\\" aria-checked=\\"false\\" aria-disabled=\\"false\\"><div class=\\"nextui-c-fwBnkL nextui-checkbox-mask\\"><i class=\\"nextui-c-iQLINm nextui-c-iQLINm-hwMtuN-indeterminate-false nextui-c-iQLINm-fHZZDy-size-xs nextui-icon-check\\"><div class=\\"nextui-c-edsSUO nextui-icon-check-line1\\"></div><div class=\\"nextui-c-khOxv nextui-icon-check-line2\\"></div></i></div></div><span class=\\"nextui-c-HDlzY nextui-c-HDlzY-cyFBRX-color-default nextui-checkbox-text\\">mini</span></label><label class=\\"nextui-c-gLvohg nextui-c-gLvohg-ewcQbF-size-sm nextui-checkbox-label\\"><div class=\\"nextui-c-hGXOEk nextui-c-huiNHE nextui-c-hGXOEk-jewhUT-color-default nextui-checkbox-container\\" tabindex=\\"0\\"><input type=\\"checkbox\\" class=\\"nextui-c-VVylU nextui-checkbox-input\\" tabindex=\\"-1\\" data-state=\\"uncheked\\" aria-checked=\\"false\\" aria-disabled=\\"false\\"><div class=\\"nextui-c-fwBnkL nextui-checkbox-mask\\"><i class=\\"nextui-c-iQLINm nextui-c-iQLINm-hwMtuN-indeterminate-false nextui-c-iQLINm-fHZZDy-size-sm nextui-icon-check\\"><div class=\\"nextui-c-edsSUO nextui-icon-check-line1\\"></div><div class=\\"nextui-c-khOxv nextui-icon-check-line2\\"></div></i></div></div><span class=\\"nextui-c-HDlzY nextui-c-HDlzY-cyFBRX-color-default nextui-checkbox-text\\">small</span></label><label class=\\"nextui-c-gLvohg nextui-c-gLvohg-daexIA-size-md nextui-checkbox-label\\"><div class=\\"nextui-c-hGXOEk nextui-c-huiNHE nextui-c-hGXOEk-jewhUT-color-default nextui-checkbox-container\\" tabindex=\\"0\\"><input type=\\"checkbox\\" class=\\"nextui-c-VVylU nextui-checkbox-input\\" tabindex=\\"-1\\" data-state=\\"uncheked\\" aria-checked=\\"false\\" aria-disabled=\\"false\\"><div class=\\"nextui-c-fwBnkL nextui-checkbox-mask\\"><i class=\\"nextui-c-iQLINm nextui-c-iQLINm-hwMtuN-indeterminate-false nextui-c-iQLINm-gXZkKf-size-md nextui-icon-check\\"><div class=\\"nextui-c-edsSUO nextui-icon-check-line1\\"></div><div class=\\"nextui-c-khOxv nextui-icon-check-line2\\"></div></i></div></div><span class=\\"nextui-c-HDlzY nextui-c-HDlzY-cyFBRX-color-default nextui-checkbox-text\\">medium</span></label><label class=\\"nextui-c-gLvohg nextui-c-gLvohg-jcUEBU-size-lg nextui-checkbox-label\\"><div class=\\"nextui-c-hGXOEk nextui-c-huiNHE nextui-c-hGXOEk-jewhUT-color-default nextui-checkbox-container\\" tabindex=\\"0\\"><input type=\\"checkbox\\" class=\\"nextui-c-VVylU nextui-checkbox-input\\" tabindex=\\"-1\\" data-state=\\"uncheked\\" aria-checked=\\"false\\" aria-disabled=\\"false\\"><div class=\\"nextui-c-fwBnkL nextui-checkbox-mask\\"><i class=\\"nextui-c-iQLINm nextui-c-iQLINm-hwMtuN-indeterminate-false nextui-c-iQLINm-gAAEng-size-lg nextui-icon-check\\"><div class=\\"nextui-c-edsSUO nextui-icon-check-line1\\"></div><div class=\\"nextui-c-khOxv nextui-icon-check-line2\\"></div></i></div></div><span class=\\"nextui-c-HDlzY nextui-c-HDlzY-cyFBRX-color-default nextui-checkbox-text\\">large</span></label><label class=\\"nextui-c-gLvohg nextui-c-gLvohg-bXPkMt-size-xl nextui-checkbox-label\\"><div class=\\"nextui-c-hGXOEk nextui-c-huiNHE nextui-c-hGXOEk-jewhUT-color-default nextui-checkbox-container\\" tabindex=\\"0\\"><input type=\\"checkbox\\" class=\\"nextui-c-VVylU nextui-checkbox-input\\" tabindex=\\"-1\\" data-state=\\"uncheked\\" aria-checked=\\"false\\" aria-disabled=\\"false\\"><div class=\\"nextui-c-fwBnkL nextui-checkbox-mask\\"><i class=\\"nextui-c-iQLINm nextui-c-iQLINm-hwMtuN-indeterminate-false nextui-c-iQLINm-gAAEng-size-xl nextui-icon-check\\"><div class=\\"nextui-c-edsSUO nextui-icon-check-line1\\"></div><div class=\\"nextui-c-khOxv nextui-icon-check-line2\\"></div></i></div></div><span class=\\"nextui-c-HDlzY nextui-c-HDlzY-cyFBRX-color-default nextui-checkbox-text\\">xlarge</span></label></div>"`;
